packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -352,32 +326,42 @@ def test_vip(dcos_api_session,
             origin_app.purge(dcos_api_session)
             log.info('Purging application: {}'.format(proxy_app.id))
             proxy_app.purge(dcos_api_session)
+            if pm_app is not None:
+                log.info('Purging application: {}'.format(pm_app.id))
+                pm_app.purge(dcos_api_session)
     assert not errors
 
 
-def setup_vip_workload_tests(dcos_api_session, container, vip_net, proxy_net, ipv6):
+def setup_vip_workload_tests(dcos_api_session, container, vip_net, proxy_net, ipv6, port_mapping_app=False):
     same_hosts = [True, False] if len(dcos_api_session.all_slaves) > 1 else [True]
-    tests = [vip_workload_test(dcos_api_session, container, vip_net, proxy_net, ipv6, named_vip, same_host)
+    tests = [vip_workload_test(dcos_api_session, container, vip_net, proxy_net,
+                               ipv6, named_vip, same_host, port_mapping_app)
              for named_vip in [True, False]
              for same_host in same_hosts]
-    for vip, hosts, cmd, origin_app, proxy_app in tests:
+    for vip, hosts, cmd, origin_app, proxy_app, pm_app in tests:
         # We do not need the service endpoints because we have deterministically assigned them
         log.info('Starting apps :: VIP: {}, Hosts: {}'.format(vip, hosts))
         log.info("Origin app: {}".format(origin_app))
         origin_app.deploy(dcos_api_session)
         log.info("Proxy app: {}".format(proxy_app))
         proxy_app.deploy(dcos_api_session)
-    for vip, hosts, cmd, origin_app, proxy_app in tests:
+        if pm_app is not None:
+            log.info("Port Mapping app: {}".format(pm_app))
+            pm_app.deploy(dcos_api_session)
+    for vip, hosts, cmd, origin_app, proxy_app, pm_app in tests:
         log.info("Deploying apps :: VIP: {}, Hosts: {}".format(vip, hosts))
         log.info('Deploying origin app: {}'.format(origin_app.id))
         origin_app.wait(dcos_api_session)
         log.info('Deploying proxy app: {}'.format(proxy_app.id))
         proxy_app.wait(dcos_api_session)
+        if pm_app is not None:
+            log.info("Deploying port mapping app: {}".format(pm_app))
+            pm_app.wait(dcos_api_session)
         log.info('Apps are ready')
     return tests
 
 
-def vip_workload_test(dcos_api_session, container, vip_net, proxy_net, ipv6, named_vip, same_host):
+def vip_workload_test(dcos_api_session, container, vip_net, proxy_net, ipv6, named_vip, same_host, port_mapping_app):
```

Comment:
maybe `is_portmap` or just `portmap` instead of `port_mapping_app` to reflect that its a bool

Reply:
To avoid conflicts with port mapping for `vip_net`/`proxy_net`, I renamed it to `with_port_mapping_app`.